src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -110,10 +116,15 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
-        handleDataSourceFetchError(
+        if (defaultDataSource) {
```

Comment:
we also need to check if the default data source has been filtered out or not

Reply:
Sure, will test this case